src/core/RenderVariables.cc

```diff
@@ -0,0 +1,22 @@
+#include "RenderVariables.h"
+#include "Context.h"
+#include "BuiltinContext.h"
+#include "Value.h"
+
+void
+RenderVariables::setRenderVariables(ContextHandle<BuiltinContext>& context) const
+{
+  context->set_variable("$preview", Value(preview));
+  context->set_variable("$t", Value(time));
+
+  auto vpr = camera.getVpr();
+  context->set_variable("$vpr", Value(VectorType(context->session(),
+    vpr.x(), vpr.y(), vpr.z())));
+  auto vpt = camera.getVpt();
+  context->set_variable("$vpt", Value(VectorType(context->session(),
+    vpt.x(), vpt.y(), vpt.z())));
+  auto vpd = camera.zoomValue();
+  context->set_variable("$vpd", Value(vpd));
+  auto vpf = camera.fovValue();
+  context->set_variable("$vpf", Value(vpf));
```

Comment:
Super nit: For all of these, if you want to be less verbose, I believe you can replace `Value(vpf)` with just `{vpf}`